src/zarr/store/memory.py

```diff
@@ -101,3 +101,63 @@ async def list_dir(self, prefix: str) -> AsyncGenerator[str, None]:
             for key in self._store_dict:
                 if key.startswith(prefix + "/") and key != prefix:
                     yield key.removeprefix(prefix + "/").split("/")[0]
+
+
+class GpuMemoryStore(MemoryStore):
```

Comment:
Maybe rename the class to something like `GpuOnlyMemoryStore` to indicate that this stores everything on the GPU no matter the original location. And point out in a docstring to use `MemoryStore` if you are location agnostic.

Reply:
Added docs about this guaranteeing storage in GPU memory and added the location agnostic clause as well. But left the name as is for simplicity.